WordPress/build.gradle

```diff
@@ -238,6 +243,33 @@ android {
     }
 }
 
+/// Dynamically add `buildConfigFields` on a given variant/flavor from prefixed properties
+/// (This is used to e.g. add every property prefixed `wp.` in `gradle.properties` as a BuildConfigField in the `wordpress` flavor)
+///
+/// `prefixes` is an array of prefixes to search for, without their `.` dot (e.g. `["wp", "jp"]`).
+/// Properties found with prefixes first in that array will be overridden by the ones found with prefixes later in the array.
+static def addBuildConfigFieldsFromPrefixedProperties(variant, properties, prefixes) {
+    // Build a Map of prefixed properties found. Keys in this Map will be lowercased and `.` replaced by `_`
+    def fields_map = [:]
+    prefixes.each { prefix ->
+        properties.each { property ->
+            if (property.key.toLowerCase().startsWith("${prefix.toLowerCase()}.")) {
+                def key = property.key.toLowerCase().replace("${prefix.toLowerCase()}.", "").replace(".", "_")
+                fields_map[key] = property.value
+            }
+        }
+    }
+
+    // Then define the found properties as buildConfigFields
+    // Also define found properties that started with `res.` (`res_*` keys once in the `fields_map`) as resValues
+    fields_map.each {
+        variant.buildConfigField "String", it.key.toUpperCase(), "\"${it.value}\""
```

Comment:
If I am not mistaken, this will currently create a `buildConfigField` for `wp.res` properties which is not what we want. If we did want to keep the `wp.res` logic (which I don't think we do), I think this would need to go into the `else` statement of the below `if`.

Reply:
This was fixed in fd55a68427af5ed079a2d3a4b8bed61cab0931e2 since we are now accepting that there will not be a `wp.res` property.

Reply:
Indeed, caught that too but figured it was not that big a deal to create the `buildConfigField` and figured I'd keep the logic as it was before my refactoring to avoid risking any side effects in case anyone relied on that.

But since we're deleting the `wp.res` logic anyway, that doesn't really matter anymore 😃